build_src.cmake

```diff
@@ -0,0 +1,179 @@
+    src/loaders/mmd3_load.c
+    src/loaders/mmd_common.c
+    src/loaders/mtm_load.c
+    src/loaders/no_load.c
+    src/loaders/okt_load.c
+    src/loaders/psm_load.c
+    src/loaders/pt3_load.c
+    src/loaders/ptm_load.c
+    src/loaders/pw_load.c
+    src/loaders/rtm_load.c
+    src/loaders/sfx_load.c
+    src/loaders/stim_load.c
+    src/loaders/st_load.c
+    src/loaders/stm_load.c
+    src/loaders/stx_load.c
+    src/loaders/sym_load.c
+    src/loaders/ult_load.c
+    src/loaders/umx_load.c
+    src/loaders/voltable.c
+    src/md5.c
+    src/med_extras.c
+    src/mix_paula.c
+    src/mkstemp.c
+    src/tempfile.c
+)
+
+set(LIBXMP_SRC_LIST_PROWIZARD
+    src/loaders/prowizard/ac1d.c
+    src/loaders/prowizard/di.c
+    src/loaders/prowizard/eureka.c
+    src/loaders/prowizard/fc-m.c
+    src/loaders/prowizard/fuchs.c
+    src/loaders/prowizard/fuzzac.c
+    src/loaders/prowizard/gmc.c
+    src/loaders/prowizard/heatseek.c
+    src/loaders/prowizard/hrt.c
+    src/loaders/prowizard/ksm.c
+    src/loaders/prowizard/mp.c
+    src/loaders/prowizard/noiserun.c
+    src/loaders/prowizard/novotrade.c
+    src/loaders/prowizard/np1.c
+    src/loaders/prowizard/np2.c
+    src/loaders/prowizard/np3.c
+    src/loaders/prowizard/p40.c
+    src/loaders/prowizard/p61a.c
+    src/loaders/prowizard/pha.c
+    src/loaders/prowizard/pm10c.c
+    src/loaders/prowizard/pm18a.c
+    src/loaders/prowizard/pp10.c
+    src/loaders/prowizard/pp21.c
+    src/loaders/prowizard/prowiz.c
+    src/loaders/prowizard/prun1.c
+    src/loaders/prowizard/prun2.c
+    src/loaders/prowizard/ptktable.c
+    src/loaders/prowizard/skyt.c
+    src/loaders/prowizard/starpack.c
+    src/loaders/prowizard/tdd.c
+    src/loaders/prowizard/theplayer.c
+    src/loaders/prowizard/titanics.c
+    src/loaders/prowizard/tp1.c
+    src/loaders/prowizard/tp3.c
+    src/loaders/prowizard/tuning.c
+    src/loaders/prowizard/unic2.c
+    src/loaders/prowizard/unic.c
+    src/loaders/prowizard/wn.c
+    src/loaders/prowizard/xann.c
+    src/loaders/prowizard/zen.c
+)
+
+set(LIBXMP_SRC_LIST_DEPACKERS
+    src/depackers/arcfs.c
+    src/depackers/bunzip2.c
+    src/depackers/crc32.c
+    src/depackers/depacker.c
+    src/depackers/gunzip.c
+    src/depackers/inflate.c
+    src/depackers/mmcmp.c
+    src/depackers/muse.c
+    src/depackers/oxm.c
+    src/depackers/ppdepack.c
+    src/depackers/readlzw.c
+    src/depackers/readrle.c
+    src/depackers/s404_dec.c
+    src/depackers/unarc.c
+    src/depackers/uncompress.c
+    src/depackers/unlha.c
+    src/depackers/unlzx.c
+    src/depackers/unsqsh.c
+    src/depackers/unxz.c
+    src/depackers/unzip.c
+    src/depackers/vorbis.c
+    src/depackers/xfnmatch.c
+    src/depackers/xz_dec_lzma2.c
+    src/depackers/xz_dec_stream.c
+)
+
+if(AMIGA)
+    list(APPEND LIBXMP_SRC_LIST_DEPACKERS
+        src/depackers/xfd.c
+        src/depackers/xfd_link.c
+    )
+endif()
+
+if(WIN32)
+    list(APPEND LIBXMP_SRC_COMMON
+        src/win32.c
```

Comment:
Technically, this file is used on other platforms as well, and should be included in `LIBXMP_SRC_LIST_LITE`.

Reply:
Yes: take this out of `if(WIN32)` and move to `LIBXMP_SRC_COMMON` and also add to `LIBXMP_SRC_LIST_LITE`

Reply:
I do see the only real case is the WATCOM usage, so, I will add the `if(WIN32 OR WATCOM)` condition, otherwise, on macOS you'll get the warning that the empty object was been linked.